tests/baselines/reference/typeFromJSInitializer.types

```diff
@@ -215,8 +215,8 @@ var n;
 
 // should get any on parameter initialisers
 function f(a = null, b = n, l = []) {
->f : (a?: null, b?: number | undefined, l?: any[]) => void
->  : ^ ^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^
+>f : (a?: null, b?: number | undefined, l?: never[]) => void

```

Comment:
`any[]` -> `never[]`? Seems like a big different in the inferred type between the syntactic builder and the inferred type. An empty `[]` is more often than not inferred as an `any[]` rather than a `never[]`. Usually for back compat.

Reply:
Reverted to old behavior.